src/billing-reports/consommation-report.component.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useState, useEffect } from 'react';
+import ReportFilterForm from './report-filter-form.component';
+import {
+  Table,
+  TableHead,
+  TableHeader,
+  TableBody,
+  TableRow,
+  TableCell,
+  Button,
+  Pagination,
+  TableExpandHeader,
+  TableExpandRow,
+  TableExpandedRow,
+  DataTable,
+  TableContainer,
+} from '@carbon/react';
+import { fetchConsommationReport, fetchAllConsommationReport } from './api/billing-reports';
+import dayjs from 'dayjs';
+import { exportSingleRecordToPDF, exportToExcel, formatValue } from './utils/download-utils';
+import styles from './billing-reports.scss';
+import { useTranslation } from 'react-i18next';
+import { formatToYMD } from './utils/download-utils';
+
+interface ReportRecord {
+  column: string;
+  value: string | string[];
+}
+
+interface ReportRow {
+  record: ReportRecord[];
+}
+
+interface Filters {
+  startDate: string;
+  endDate: string;
+  company: string;
+}
+
+const ConsommationReport: React.FC = () => {
+  const { t } = useTranslation();
+
+  const [results, setResults] = useState<ReportRow[]>([]);
+  const [columns, setColumns] = useState<string[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(50);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [currentFilters, setCurrentFilters] = useState<Filters | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const recordMap = new Map<string, ReportRecord[]>();
```

Comment:
recordMap object is re-created on every render which will slow down the UI especially if the dataset is large. I suggest you use React’s `useMemo` to memoize expensive computations based on their dependencies e.g. memoize the mapping of results to rows, or the creation of recordMap and remove any in-render mutations like recordMap.set(...).

```suggestion
const recordMap = useMemo(() => {
  const map = new Map();
  results.forEach((row, idx) => {
    const id = `${(page - 1) * pageSize + idx + 1}`;
    map.set(id, row.record);
  });
  return map;
}, [results, page, pageSize]);
```